backend/src/interactions/updateUnconfirmedContribution/UnconfirmedContributionUserAddMessage.role.ts

```diff
@@ -0,0 +1,51 @@
+import { Contribution } from '@entity/Contribution'
+import { User } from '@entity/User'
+
+import { ContributionMessageBuilder } from '@/data/ContributionMessage.builder'
+import { ContributionMessageArgs } from '@/graphql/arg/ContributionMessageArgs'
+import { ContributionStatus } from '@/graphql/enum/ContributionStatus'
+import { LogError } from '@/server/LogError'
+
+import { AbstractUnconfirmedContributionRole } from './AbstractUnconfirmedContribution.role'
+
+export class UnconfirmedContributionUserAddMessageRole extends AbstractUnconfirmedContributionRole {
+  public constructor(contribution: Contribution, private updateData: ContributionMessageArgs) {
+    super(contribution, contribution.amount, contribution.contributionDate)
+  }
+
+  protected update(): void {
+    if (
+      this.self.contributionStatus === ContributionStatus.IN_PROGRESS ||
+      this.self.resubmissionAt !== null
+    ) {
+      this.self.contributionStatus = ContributionStatus.PENDING
+      this.self.resubmissionAt = null
+    } else {
+      this.changed = false
+    }
+  }
+
+  protected checkAuthorization(user: User): AbstractUnconfirmedContributionRole {
+    if (this.self.userId !== user.id) {
+      throw new LogError('Can not update contribution of another user', this.self, user.id)
+    }
+    // only admins and moderators can update it when status is other than progress or pending
+    if (
+      this.self.contributionStatus !== ContributionStatus.IN_PROGRESS &&
+      this.self.contributionStatus !== ContributionStatus.PENDING
+    ) {
```

Comment:
following the comment i miss the check on user for having admin or moderator role. there is only a check on the contributionStatus.

Reply:
i only ask, because the comment suggerates that the if-claus will check for 'user-role' AND 'status', but the if-clause only contains a check for `status`.
If this leads to the requested business behaviour is for sure an other question, we could discuss outside this PR.

Reply:
ok, I add additional comment line to clarify it.